### ISSUE TITLE
Register prompt-radio as a generate plugin ?

### ISSUE THREAD

**Issue by rbecheras** [enquirer/enquirer-prompt-radio#3]
Is possible to use `prompt-radio` as a `Generte` plugin ?

For now I fail to do it:

## generator.js

```js
import isValid from 'is-valid-app'

import Logger from './utils/Logger'
import { task } from './utils/utils'

import promptRadio from 'prompt-radio'

import generateDefaults from 'generate-defaults'
import generateDest from 'generate-dest'
import generatePackage from './subgenerators/generate-swap-package/generator'

// import promptTask from './tasks/prompt'

const log = new Logger('generate-swap-project')

export default function (app) {
  if (!isValid(app, 'generate-swap-project')) return

  app.on('error', ::log.error)

  /**
   * Plugins
   */
  app.use(generateDefaults)
  app.use(promptRadio)

  /**
   * Micro generators (as plugins)
   */
  app.register('destination-directory', generateDest)
  app.register('package', generatePackage)

  /**
   * Scaffold out a(n) swap-project project. Also aliased as the [default](#default) task.
   *
   * ```sh
   * $ gen swap-project:project
   * ```
   * @name project
   * @api public
   */

  app.task('project', function (cb) {
    app.generate([
      'prompt',
      'destination-directory:default',
      'package',
      'testfile'
    ], cb)
  })

  // app.task('prompt', promptTask(app))
  app.task('prompt', cb => {
    app.question({
      name: 'githost',
      message: 'Git remote host ?',
      type: 'radio',
      default: 'github.com',
      choices: ['github.com', 'gitlab.sirap.fr', 'gitlab.com']
    })
    app.ask(['githost'])
  })

  /**
   * Write a `package.json` file to the current working directory.
   * Call the `package:default` task from the sub generator `generate-swap-package`.
   *
   * ```sh
   * $ gen swap-project:package
   * ```
   * @name package
   * @api public
   */
  app.task('package', function (cb) {
    app.generate(['package:default'], cb)
  })

  /**
   * Write a `test-file.txt` file to the current working directory.
   *
   * ```sh
   * $ gen swap-project:file
   * ```
   * @name file
   * @api public
   */
  task(app, 'testfile', 'test-file.txt')

  /**
   * Scaffold out a new swap-project project. This task is an alias for the [swap-project](#swap-project)
   * task, to allow running this generator with the following command:
   *
   * ```sh
   * $ gen swap-project
   * ```
   * @name default
   * @api public
   */
  app.task('default', ['project'])
}
```

Then running it:

```
$ gen swap-project
[00:54:10] starting generate
[00:54:11] ✔ running tasks: [ 'swap-project' ]
/home/remi/d/dev/generate-swap-project/node_modules/prompt-base/index.js:72
  this.ui.rl.pause();
             ^

TypeError: Cannot read property 'pause' of undefined
    at Prompt (/home/remi/d/dev/generate-swap-project/node_modules/prompt-base/index.js:72:14)
    at Generate.Prompt (/home/remi/d/dev/generate-swap-project/node_modules/prompt-base/index.js:44:12)
    at Generate.Checkbox (/home/remi/d/dev/generate-swap-project/node_modules/prompt-checkbox/index.js:13:10)
    at Generate.Radio (/home/remi/d/dev/generate-swap-project/node_modules/prompt-radio/index.js:12:12)
    at Generate.use (/home/remi/.nvm/versions/node/v8.9.3/lib/node_modules/generate/node_modules/base-plugins/index.js:95:18)
    at Generate.exports.default (/home/remi/d/dev/generate-swap-project/dist/lib/generator.js:22:7)
    at Env.invoke (/home/remi/.nvm/versions/node/v8.9.3/lib/node_modules/generate/node_modules/base-env/lib/env.js:118:23)
    at Generate.<anonymous> (/home/remi/.nvm/versions/node/v8.9.3/lib/node_modules/generate/node_modules/base-generators/lib/generator.js:62:30)
    at Generate.getGenerator (/home/remi/.nvm/versions/node/v8.9.3/lib/node_modules/generate/node_modules/base-generators/index.js:157:22)
    at Object.exports.process (/home/remi/.nvm/versions/node/v8.9.3/lib/node_modules/generate/node_modules/base-generators/lib/tasks.js:127:24)
    at /home/remi/.nvm/versions/node/v8.9.3/lib/node_modules/generate/node_modules/base-generators/lib/tasks.js:86:23
    at Array.reduce (<anonymous>)
    at exports.parse (/home/remi/.nvm/versions/node/v8.9.3/lib/node_modules/generate/node_modules/base-generators/lib/tasks.js:85:15)
    at Generate.generate (/home/remi/.nvm/versions/node/v8.9.3/lib/node_modules/generate/node_modules/base-generators/index.js:490:21)
    at /home/remi/.nvm/versions/node/v8.9.3/lib/node_modules/generate/node_modules/base-generators/index.js:475:18
    at iterate (/home/remi/.nvm/versions/node/v8.9.3/lib/node_modules/generate/node_modules/async-each-series/index.js:8:5)

```

May it is not possible ?

How to use enquirer prompts in my generator ?

**Comment by jonschlinkert** [comment]
No, the prompts should be used separately. The readme's for each prompt show how to use them. There is also an `example` directory in each project.

**Comment by rbecheras** [comment]
Thanks for answering me @jonschlinkert,

So, to use custom prompts (radio, checkbox, list...) in a `Generate` generator, the recommended way is to `require('enquirer')` and for example `prompt-radio` in the `generator.js` ? Or is there an other dedicated way to custom prompt for `Generate/Update/Base` generators/plugins ?

**Comment by jonschlinkert** [comment]
sorry for the late reply, I didn't see your last question @rbecheras. did you get this figured out?

**Comment by doowb** [comment]
Closing as this prompt has been completely refactored from the ground up, and is now included in [Enquirer 2.0](https://github.com/enquirer/enquirer). Please direct all questions, issues and feature requests to that repository.
